solr/test-framework/src/jmh/org/apache/solr/bench/DocMakerRamGen.java

```diff
@@ -0,0 +1,269 @@
+      } else {
+        return TestUtil.randomRealisticUnicodeString(new Random(seed), 1, value.getMaxLength());
+      }
+    }
+
+    if (value.getLength() > -1) {
+      return TestUtil.randomRealisticUnicodeString(ThreadLocalRandom.current(), value.getLength(), value.getLength());
+    } else {
+      return TestUtil.randomRealisticUnicodeString(ThreadLocalRandom.current(), 1, value.getMaxLength());
+    }
+  }
+
+  private String getAlphabeticString(FieldDef value) {
+    if (value.getMaxCardinality() > 0) {
+      long start = value.getCardinalityStart();
+      long seed = nextLong(start, start + value.getMaxCardinality(), threadRandom);
+      SplittableRandom random = new SplittableRandom(seed);
+      if (value.getLength() > -1) {
+        return RandomStringUtils.random(nextInt(value.getLength(), value.getLength(), random), 0, 0, true, false, null, new Random(seed));
+      } else {
+        return RandomStringUtils.random(nextInt(1, value.getMaxLength(), random), 0, 0, true, false, null, new Random(seed));
+      }
+    }
+
+    SplittableRandom threadRandom = new SplittableRandom(Long.getLong("threadLocalRandomSeed", ThreadLocalRandom.current().nextLong()));
+    if (value.getLength() > -1) {
+      return RandomStringUtils.random(nextInt(value.getLength(), value.getLength(), threadRandom), 0, 0, true, false, null, ThreadLocalRandom.current());
+    } else {
+      return RandomStringUtils.random(nextInt(1, value.getMaxLength(), threadRandom), 0, 0, true, false, null, ThreadLocalRandom.current());
+    }
+  }
+
+  public enum Content {
+    UNICODE, ALPHEBETIC, INTEGER, UNIQUE_INT
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    DocMakerRamGen that = (DocMakerRamGen) o;
+    return fields.equals(that.fields);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(fields);
+  }
+
+  public static int nextInt(final int startInclusive, final int endExclusive, SplittableRandom random) {
+    Validate.isTrue(endExclusive >= startInclusive,
+        "Start value must be smaller or equal to end value.");
+    Validate.isTrue(startInclusive >= 0, "Both range values must be non-negative.");
+
+    if (startInclusive == endExclusive) {
+      return startInclusive;
+    }
+
+    return startInclusive + random.nextInt(endExclusive - startInclusive);
+  }
+
+  public static long nextLong(final long startInclusive, final long endExclusive, SplittableRandom random) {
+    Validate.isTrue(endExclusive >= startInclusive,
+        "Start value must be smaller or equal to end value.");
+    Validate.isTrue(startInclusive >= 0, "Both range values must be non-negative.");
+
+    if (startInclusive == endExclusive) {
+      return startInclusive;
+    }
+
+    return startInclusive + random.nextLong(endExclusive - startInclusive);
+  }
+
+  public static void main(String[] args) {
```

Comment:
This method should perhaps not be kept?

Reply:
Still draft form. This should end up as a unit test for the jmh stuff, but have not set that ability up in gradle quite yet.